.github/workflows/deploy-2-start.yml

```diff
@@ -29,12 +29,23 @@ jobs:
       - uses: actions/checkout@v4
         with:
           ref: ${{ inputs.version }}
+      - name: Setup ssh
+        run: |
+          mkdir -p ~/.ssh
+          chmod 700 ~/.ssh
+          echo '${{ secrets.SSH_KEY }}' > ~/.ssh/priv.key
+          chmod 600 ~/.ssh/priv.key
+          ssh-keyscan ${{ secrets.HOST }} >> ~/.ssh/known_hosts
+          ssh-keyscan ${{ secrets.HOST_DATA }} >> ~/.ssh/known_hosts
+          chmod 600 ~/.ssh/known_hosts 
       - name: Copy spack.yaml
-        run: rsync spack.yaml ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.SPACK_YAML_LOCATION }}
+        run: rsync -e 'ssh -i ~/.ssh/priv.key' \
```

Comment:
An alternative to adding to known hosts might to setup an ~/.ssh/config with Hostname, User, IdentityFile etc so subsequent rsync/ssh cmds could just use the Host  rather than `${{ secrets.USER}}@${{ secrets.HOST }} -i ~/.ssh/deploy.priv `. In this case I think it will still add the same about of code so might not be worth it

Reply:
Thats a good idea - I'll create an issue for it. Thanks for all your help with the development of this pull request :)